### PR TITLE
feat(metric): add text value

### LINKED ISSUE #1801: [Metric] support strings
**Is your feature request related to a problem? Please describe.**
We would like to support string values in the Lens metric visualization in Kibana. One use case is displaying the last value of a server status field: "up" or "down."

**Describe the solution you'd like**
The metric should accept a string value as well as a numeric value for the primary metric. Since a non-numeric value doesn't make sense in the context of a progress bar or a trend line, this probably means a new type of metric datum interface that can accept a string value, while the others maintain the numeric constraint.

**Describe alternatives you've considered**
I don't see any alternative.

**Kibana Cross Issues**
https://github.com/elastic/kibana/issues/137756

**Checklist**

- [x] this request is checked against already exist requests
- [x] every related Kibana issue is listed under `Kibana Cross Issues` list
- [x] `kibana cross issue` tag is associated to the issue if any kibana cross issue is present

### PR DESCRIPTION
**Pull request by markov00**
## Summary

Add the ability to render text in place of numerical values. 

### BREAKING CHANGE

The metric types are updated to align with the new feature. I've introduced a lower type branch, that splits numerical (together with trend/progress) from a textual only metric. 

Screenshot 2022-09-12 at 14 02 33

## Issues

fix #1801

### Checklist

- [x] The proper **chart type** label has been added (e.g. `:xy`, `:partition`)
- [x] All related issues have been linked (i.e. `closes #123`, `fixes #123`)
- [x] New public API exports have been added to `packages/charts/src/index.ts`
- [ ] Unit tests have been added or updated to match the most common scenarios
- [ ] The proper documentation and/or storybook story has been added or updated
- [ ] The code has been checked for cross-browser compatibility (Chrome, Firefox, Safari, Edge)
- [ ] Visual changes have been tested with all available themes including `dark`, `light`, `eui-dark` & `eui-light`

### CODE REVIEW

File: packages/charts/api/charts.api.md [hunk 1]
```diff
@@ -1718,8 +1718,6 @@ export const Metric: FC<SFProps<MetricSpec, "chartType" | "specType", "data", ne
```
Comment (nickofthyme) [inline packages/charts/api/charts.api.md:1718]: Should we be truncating the value? Be it a string or a formatted number?

We could always truncate and put the full value in the `title` prop. This should be standard practice for native DOM UI.

Screen Recording 2022-09-13 at 09 02 56 AM

**Review by nickofthyme — APPROVED** [review]
Looks good with one thought.

File: packages/charts/api/charts.api.md (hunk 1 above)
Comment (markov00) [reply packages/charts/api/charts.api.md:1718]: you are right, let me add that to the title

Comment (markov00) [reply packages/charts/api/charts.api.md:1718]: I've added the ellipsis to the text truncation in https://github.com/elastic/elastic-charts/pull/1817/commits/b46b7e18206ae420c6f5197bcb1c801b7f2f8e87 and fixed to have it only for textual values in https://github.com/elastic/elastic-charts/pull/1817/commits/b066345cd53672dff12ea6404a4a54e739a06551

File: packages/charts/src/chart_types/metric/renderer/dom/text.tsx [hunk 2]
```diff
@@ -221,19 +234,25 @@ export const MetricText: React.FunctionComponent<{
...
+                    text
+                  ) : (
+                    // eslint-disable-next-line react/no-array-index-key
+                    <span
+                      key={`${text}${i}`}
+                      className="echMetricText__part"
+                      style={{ fontSize: `${VALUE_PART_FONT_SIZE[size]}px` }}
+                    >
+                      {text}
+                    </span>
```
Comment (nickofthyme) [inline packages/charts/src/chart_types/metric/renderer/dom/text.tsx:251]: Add the `title` prop to the value, assuming the `text` here is the full text.

Screen Recording 2022-09-14 at 08 53 34 AM


```suggestion
                    <span
                      key={`${text}${i}`}
                      title={text}
                      className="echMetricText__part"
                      style={{ fontSize: `${VALUE_PART_FONT_SIZE[size]}px` }}
                    >
                      {text}
                    </span>
```

**Review by nickofthyme — APPROVED** [review]
Good, one finally suggestion

File: packages/charts/src/chart_types/metric/renderer/dom/text.tsx (hunk 2 above)
Comment (markov00) [reply packages/charts/src/chart_types/metric/renderer/dom/text.tsx]: great, done here: https://github.com/elastic/elastic-charts/pull/1817/commits/0ab47281cc271d6816370d1b7e60897e4857d42b
I've also cleaned up a bit the code

File: packages/charts/src/chart_types/metric/renderer/dom/text.tsx [hunk 3]
```diff
@@ -221,36 +241,45 @@ export const MetricText: React.FunctionComponent<{
...
+            );
+          })}
         </p>
       </div>
     </div>
   );
 };
 
-function splitNumericSuffixPrefix(text: string) {
+function splitNumericSuffixPrefix(text: string): ['normal' | 'small', string][] {
```
Comment (nickofthyme) [inline packages/charts/src/chart_types/metric/renderer/dom/text.tsx:271]: Can you also add the tuple values here?

```suggestion
function splitNumericSuffixPrefix(text: string): [thing1: 'normal' | 'small', thing2: string][] {
```

**Review by nickofthyme — APPROVED** [review]

**Comment by markov00** [comment]
buildkite update screenshots

File: packages/charts/src/chart_types/metric/renderer/dom/text.tsx (hunk 3 above)
Comment (markov00) [reply packages/charts/src/chart_types/metric/renderer/dom/text.tsx]: solved in https://github.com/elastic/elastic-charts/pull/1817/commits/05b71181503948465dbf58f03aa82fafa1da14c8

**Pull request merged**